### DISCUSSION TITLE
Is it possible render markdown inside HTML Blocks?

### DISCUSSION THREAD

**Discussion by jasonvarga** [thephpleague/commonmark#634] [Q&A] [answered]
## Question

I've tried my best to search for the answer, so apologies if this has already been answered.

Is there a way to render markdown inside html blocks?

Basically, this feature from Markdown Extra: https://michelf.ca/projects/php-markdown/extra/#markdown-attr

By adding `markdown="1"` to a div, it would parse inside of it.

```
# Heading

<div markdown="1">
# Heading
</div>
```

Right now, markdown inside html just gets output as-is:

```
<h1>Heading</h1>
<div markdown="1">
# Heading
</div>
```

I understand this would likely need to be an extension.

**Comment by markhalliwell** [comment, marked as answer]
This is intentional per the [commonmark spec](https://spec.commonmark.org/0.29/#html-blocks). You need to include a blank line after the opening HTML tag and before the closing tag:

```markdown
# Heading

<div>

# Heading

</div>
```

Will become: 
```html
<h1>Heading</h1>
<div>
<h1>Heading</h1>
</div>
```

https://johnmacfarlane.net/babelmark2/?text=%23+Heading%0A%0A%3Cdiv%3E%0A%0A%23+Heading%0A%0A%3C%2Fdiv%3E

**Comment by jasonvarga** [comment]
Oh awesome, thanks.
